umad-02-keys-and-authentication.md

```diff
@@ -25,42 +25,56 @@ section.
 
 ## Public Key Exchange
 
-VASPs expose their public keys to other VASPs by responding to `GET` requests at the endpoint
+TThe UMA protocol relies on X.509 certificates for public key exchange among VASPs. In order to obtain an X.509
+certificate, the VASP is required to submit a request to a trusted certificate authority (CA), which issues the
+certificate. The issuing CA must keep track of certificates that are revoked, and provide this information to other
+parties via Certificate Revocation Lists (CRLs) or an Online Certificate Status Protocol (OCSP) server. The URLs for
+accessing a CA's CRL/OCSP can be found inside the certificate, and VASPs should ensure the validity of all of the
+certificates they receive to ensure compliance and security.
+
+It is not recommended for VASPs to use self-signed certificates when communicating with other VASPs outside their
+organization. They should only be used in test environments. To generate a self-signed X.509 certificate based on the
+key generated above, run:
+
+```bash
+# Generate a certificate based on the key we generated:
+$ openssl req -new -x509 -key ec_key.pem -sha256 -nodes -out ec_crt.crt -days <expiration in days>
+
+# Print out the certificate data:
+$ openssl x509 -in ec_crt.crt -noout -text
+```
+
+VASPs expose their certificates to other VASPs by responding to `GET` requests at the endpoint
 `https://<vaspdomain>/.well-known/lnurlpubkey`. This endpoint returns a JSON object with the following structure:
 
 ```json
```

Comment:
Do we want to include the existing v0 fields for backwards-compatibility? I guess maybe not in the spec, but maybe in example vasp implementations? wdyt?